java/vector/src/main/codegen/templates/LegacyUnionVector.java

```diff
@@ -0,0 +1,339 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+<@pp.dropOutputFile />
+<@pp.changeOutputFile name="/org/apache/arrow/vector/complex/UnionVector.java" />
```

Comment:
This is in the new vector class space. @siddharthteotia, what are your thoughts on:

* Do we want to keep codegen union vector in the new vector package? 

* Are there any API changes comparing new/legacy union vectors?

* Do we want to release support for union vector in new vector classes in 0.8?

Reply:
Oh, I see this is acutally generate `LegacyUnionVector.java` - the doc is incorrect.

To answer myself, it seems:
* Yes
* (sid can you answer this?)
* Yes it's compatible with the C++ union. No otherwise.